Loop/Managers/Alerts/AlertManager.swift

```diff
@@ -318,15 +351,29 @@ extension AlertManager: AlertIssuer {
             modalAlertIssuer?.issueAlert(alert)
         }
     }
+
+    private func issueAlertWithHandlers(_ alert: Alert, issuedDate: Date = Date()) {
+        // alert may need to be muted
+        let processedAlert = alertMuter.processAlert(alert, issuedDate: issuedDate)
```

Comment:
It would seem clearer to me to have the alert issuers know about the current mute state and do the right thing when issuing, rather than mutating alerts to mark them as muted or not.  This is reflecting my assumption that the "muting" we're implementing here is a property of the alert system, not a property of individual alerts.

Reply:
I started down this path but it requires a change to the AlertIssuer protocol that doesn't make sense for all AlertIssuers (cgm managers, pump managers, loop device manager, etc). All of those do not care of the system mute state, just that an alert needs to be issued. Only the `AlertManager` goes through the act of muting an alert. From our discussion of a system property, I thought this was in line.

Please suggest an alternative.

Reply:
Yeah, I definitely agree about not changing the protocols used by sources of alerts (device managers, etc), including AlertIssuer.  I'm suggesting a change in the opposite direction, I think; I'm suggesting that the Modal alert presenter and the system notification presenter are made aware of the mute state, and do the right thing, rather than having to tweak the alerts themselves, as is being done here. 

I'm guessing maybe you are hitting difficulties with AlertManager accessing `UserNotificationAlertIssuer` and `InAppModalAlertIssuer` via the `AlertIssuer` protocol. This has proved constraining to me in the past as well, and I think it might be worth revisiting how that relationship is modeled. Those two objects are really core parts of our alert system; they're not just generic "AlertIssuers" in the same sense of how devices use that api to issue alerts. I feel the relationship between AlertManager and those two classes is closer, and that representing that relationship as an unbounded array  (`handlers`) of `AlertIssuers` is maybe a premature abstraction.  When I tried to replace that with direct references, however, updating the tests proved difficult. You'll probably hit the same thing if you try it, but I'd welcome any steps in that direction.